v2/internal/frontend/desktop/common/process_request.go

```diff
@@ -0,0 +1,55 @@
+package common
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/wailsapp/wails/v2/internal/frontend/assetserver"
+)
+
+type RequestRespone struct {
+	Body       []byte
+	MimeType   string
+	StatusCode int
+}
+
+func (r RequestRespone) StatusText() string {
+	return http.StatusText(r.StatusCode)
+}
+
+func (r RequestRespone) String() string {
+	return fmt.Sprintf("Body: '%s', StatusCode: %d", string(r.Body), r.StatusCode)
+}
+
+func ProcessRequest(uri string, assets *assetserver.DesktopAssetServer, expectedScheme string, expectedHosts ...string) (RequestRespone, error) {
+	// Translate URI to file
+	file, err := translateUriToFile(uri, expectedScheme, expectedHosts...)
+	if err != nil {
+		if err == ErrUnexpectedHost {
+			body := fmt.Sprintf("expected host one of \"%s\"", strings.Join(expectedHosts, ","))
+			return textResponse(body, http.StatusInternalServerError), err
+		}
+
+		return RequestRespone{StatusCode: http.StatusInternalServerError}, err
```

Comment:
I would have thought that this should be 400/Bad Request? From the point of view of the backend server..

Reply:
Yeah, that makes absolutely sense.